### ISSUE TITLE
Is it possible to turn off motion/animation for all components?

### ISSUE THREAD

**Issue by coodoo** [chakra-ui/chakra-ui#3696]
Currently there's `motionPreset` for `Modal`, it would be great if there's a global switch to turn off all motion/animation effects for every component, any thought?

**Comment by vickOnRails** [comment]
Hey there @coodoo 👋🏼

First off, I didn't find a perfect way to turn off animations on all components. 

But I took a little time to look further. I've found there are 2 ways you might want to go about this
- `usePrefersReducedMotion` custom hook ([here](https://chakra-ui.com/docs/hooks/use-prefers-reduced-motion))
- CSS [Reduced Motion](https://css-tricks.com/introduction-reduced-motion-media-query/) Media Query

The first approach would involve you obtaining and storing the user's motion preference at some top-level place in your app.

```jsx
import {  usePrefersReducedMotion } from "@chakra-ui/react";

export default App() {
  // You can then store this value in some state or localstorage
  const prefersMotion = usePrefersReducedMotion();
}
```

You'll then have to use it in specific components that allow you to turn on/off motion (Like in your modal example)

```jsx

import React from "react";
import ReactDOM from "react-dom";
import {
  Button,
  Modal,
  ChakraProvider,
  usePrefersReducedMotion
} from "@chakra-ui/react";

export function App() {
  const prefersReducedMotion= usePrefersReducedMotion();

  const { isOpen, onOpen, onClose } = useDisclosure();
  return (
    <ChakraProvider>
      <Button onClick={onOpen}>Trigger modal</Button>

      <Modal
        onClose={onClose}
        isOpen={isOpen}
        isCentered
        motionPreset={prefersReducedMotion ? "none" : "slideInBottom"}
      >
        {...ModalContent}
      </Modal>
    </ChakraProvider>
  );
}
```
I've put together a working demo [here](https://6913w.csb.app/)

> You'll have to turn off/on motion on your OS to test

The only downside to this approach is that you can't turn it off globally (Not that I know of)
--- 

- The second approach I thought of didn't work. It would involve using the native CSS [prefers motion](https://css-tricks.com/introduction-reduced-motion-media-query/) query and turning off transitions on every element on the page. 

```css
@media screen and (prefers-reduced-motion) {
  * {
    /* 🤷🏼‍♀️ */
    transition: none !important;
  }
}
```

I'll update this issue as I learn more. Or @segunadebayo, is there a way around this already?

**Comment by segunadebayo** [comment]
Thanks for chiming in @vickOnRails 👌.

We have this in our roadmap as it's a commonly requested feature. I'll land a PR for this pretty soon

**Comment by ashot** [comment]
any update on this?

**Comment by bryantobing12** [comment]
any update?

**Comment by fandy** [comment]
The Menu animation is very janky. Would be great if there was a way to turn it off.

**Comment by vladnicula** [comment]
Got here via google for the same menu animation problem mentioned by fandy. Is there any way to hack the animation out of existence for the menu?

**Comment by fandy** [comment]
@segunadebayo is the animation configuration on a public roadmap somewhere?

**Comment by youminkim** [comment]
Modal open/close animation is pretty laggy from recent iPhone. Turning off animation would be great

**Comment by bymoe** [comment]
I think some people -me included- wants this option not because of disabling animations but because of the framer-motion huge dependency that most of the time useless just for few unnoticeable animations which can be made with simple css :man_shrugging:

**Comment by Bertg** [comment]
Something to consider for v2, possibly v2.1? @segunadebayo

**Comment by Mehenni-B** [comment]
any update?

**Comment by benjamincharity** [comment]
Bump

**Comment by billgloff** [comment]
Any updates on this?

**Comment by Bertg** [comment]
Can we add the `Topic: Testing 🧪` label to this issue as well? @segunadebayo
This is rather relevant to that as well.

**Comment by jhleao** [comment]
Bump. Would be great to have this ability. Slow animations in lower end devices tend to increase perceived lag a lot

**Comment by JaapWeijland** [comment]
Bump! I'd like to change the menu animation!

**Comment by JTBrinkmann** [comment]
@billgloff @Bertg @JaapWeijland @benjamincharity As a general advice for forums/Github: instead of "bump" or "any updates?" comments, you can subscribe to this issue to get notified about updates and click/tap the :+1: below the top comment. Bumping won't make things magically get more prioritised. I hope this won't come across as rude, that's not my intention.

**Comment by benjamincharity** [comment]
@JTBrinkmann apologies, but afaik subscribing doesn't stop stale bots etc from closing out valid issues.

**Comment by Bertg** [comment]
@JTBrinkmann I do think I acted as a good comment member here. I think each of my comment was valid at the time it was made. I am subscribed and I clicked the 👍-button.

And like @benjamincharity said, stale bot only responds to comments...

Also, I just noticed work started on this; which is great (which we didn't get updated about, even though I am subscribed). 
So for those monitoring this ticket: Good news! Work started here: https://github.com/chakra-ui/chakra-ui/pull/6844 🥳

**Comment by adam-thomas-privitar** [comment]
With the latest framer motion you can wrap your app in 

```
<MotionConfig reducedMotion="user">{children}</MotionConfig>
```

To respect users accessibility settings. Or to force off:

```
<MotionConfig reducedMotion="always">{children}</MotionConfig>
```

**Comment by david-morris** [comment]
@adam-thomas-privitar great tip!
I am still wondering why `reducedMotion="user"` isn't the default, when other parts of Chakra-UI are designed to be accessible from the start, like IconButton requiring an aria-label in the type.

**Comment by adam-thomas-privitar** [comment]
I forgot to add though, it doesn't remove all animations. It selectively reduces certain ones. Framer should add an option for total removal see - https://github.com/framer/motion/issues/1160.

However, if doing this for testing, the reduced motion reduced or even eliminated certain flakes for me.

**Comment by penx** [comment]
Came across this today with the Chakra checkbox. It seems that a checked checkbox will always animate in on first render, can't be configued, and doesn't respect prefer-reduced-motion.

https://github.com/chakra-ui/chakra-ui/blob/eb7cab94cc6e762e96c61c63c17d3414b9c71ed6/packages/components/checkbox/src/checkbox.tsx#L137-L144

> This is especially handy when dealing with tools like Storybook and Chromatic

This is also how I ended up here :)

Not sure how to deal with this other than setting a delay to _every_ story that contains a checkbox.

I'm assuming a PR to make Checkbox respect prefer-reduced-motion would be welcomed?

**Comment by jmcoder1** [comment]
I would like to disable framer-motion at lower breakpoints

**Comment by kul-sudo** [comment]
Any updates?

**Comment by kul-sudo** [comment]
Are there any updates or new workarounds for this flaw? Has it been fixed?

**Comment by c-semaan** [comment]
Bump, we need this the menu animations are janky

**Comment by adam-maj** [comment]
After much digging, I finally figured out a complete solution! Leaving it here for anyone who finds this thread in the future - turns out it's all possible already 😄

There's a `motionProps` field on the `<MenuList />` component that lets you completely edit the animations. You can see [in the `MenuList` code](https://github.com/chakra-ui/chakra-ui/blob/7b6e66a15b08ad27e8458a009c3fb15ee738ca37/packages/components/src/menu/menu-list.tsx#L68) that the `motionProps` overrides all other animation config (here are the [default animations](https://github.com/chakra-ui/chakra-ui/blob/7b6e66a15b08ad27e8458a009c3fb15ee738ca37/packages/components/src/menu/menu-list.tsx#L19) written by the Chakra team).

I got the behavior I wanted by overriding this field as follows (I still wanted a fade-in/fade-out animation, but I just don't like the ugly menu size increase/decrease transition)

```tsx
<MenuList
  motionProps={{
    // Configure the initial state pre-transition (optional)
    initial: {
      visibility: "hidden",
      opacity: 0,
      scale: 1,
    },
    variants: {
      enter: {
        visibility: "visible",
        opacity: 1,
        scale: 1,
        transition: {
          // If you want to completely get rid of the animation, just change this duration to 0
          duration: 0.3,
          ease: [0.4, 0, 0.2, 1],
        },
      },
      exit: {
        transitionEnd: {
          visibility: "hidden",
        },
        opacity: 0,
        scale: 1,
        transition: {
          // Same with this duration on exit animation
          duration: 0.1,
          easings: "easeOut",
        },
      },
    },
  }}
>
  {children}
</MenuList>
```

Hope this helps people!

**Comment by adam-maj** [comment]
> After much digging, I finally figured out a complete solution! Leaving it here for anyone who finds this thread in the future - turns out it's all possible already 😄
> 
> There's a `motionProps` field on the `<MenuList />` component that lets you completely edit the animations. You can see [in the `MenuList` code](https://github.com/chakra-ui/chakra-ui/blob/7b6e66a15b08ad27e8458a009c3fb15ee738ca37/packages/components/src/menu/menu-list.tsx#L68) that the `motionProps` overrides all other animation config (here are the [default animations](https://github.com/chakra-ui/chakra-ui/blob/7b6e66a15b08ad27e8458a009c3fb15ee738ca37/packages/components/src/menu/menu-list.tsx#L19) written by the Chakra team).
> 
> I got the behavior I wanted by overriding this field as follows (I still wanted a fade-in/fade-out animation, but I just don't like the ugly menu size increase/decrease transition). You can also completely get rid of the animations with this method by changing the durations to 0.
> 
> ```tsx
> <MenuList
>   motionProps={{
>     // Configure the initial state pre-transition (optional)
>     initial: {
>       visibility: "hidden",
>       opacity: 0,
>       scale: 1,
>     },
>     variants: {
>       enter: {
>         visibility: "visible",
>         opacity: 1,
>         scale: 1,
>         transition: {
>           // If you want to completely get rid of the animation, just change this duration to 0
>           duration: 0.3,
>           ease: [0.4, 0, 0.2, 1],
>         },
>       },
>       exit: {
>         transitionEnd: {
>           visibility: "hidden",
>         },
>         opacity: 0,
>         scale: 1,
>         transition: {
>           // Same with this duration on exit animation
>           duration: 0.1,
>           easings: "easeOut",
>         },
>       },
>     },
>   }}
> >
>   {children}
> </MenuList>
> ```
> 
> Hope this helps people!

cc @segunadebayo on resolution

**Comment by ehynds** [comment]
This is now supported by `framer-motion` in `v10.17.0`: https://github.com/framer/motion/pull/2461

```tsx
import { MotionGlobalConfig } from "framer-motion"

MotionGlobalConfig.skipAnimations = true
```

**Comment by NASlR** [comment]
```
MotionGlobalConfig.skipAnimations = true
```
Doesn't work for the transition components <Fade /> etc. This is because the version of framer-motion in the components package is below v10.17.0

I've upgraded it and written a test in https://github.com/chakra-ui/chakra-ui/pull/8354

Hopefully we can get this resolved now!

**Comment by segunadebayo** [comment]
Closing based on https://github.com/chakra-ui/chakra-ui/issues/3696#issuecomment-1965451264 and https://github.com/chakra-ui/chakra-ui/issues/3696#issuecomment-1303253461

This is out of scope for v2 at the moment. We're focusing on launching v3 instead.

**Comment by MaxwellDG** [comment]
```
motionProps={{
    initial: false 
}}
```
